src/shared/telemetries/telemetrysender/sender.go

```diff
@@ -55,6 +70,58 @@ func (t *TelemetrySender) Send(component telemetriesgql.Component, eventType tel
 
 	telemetryData := telemetriesgql.TelemetryData{EventType: eventType, Count: count}
 	telemetry := telemetriesgql.TelemetryInput{Component: component, Data: telemetryData}
-	t.batcher.AddNoWait(telemetry)
+	t.eventBatcher.AddNoWait(telemetry)
+
+	return nil
+}
+
+func (t *TelemetrySender) IncrementCounter(component telemetriesgql.Component, eventType telemetriesgql.EventType, key string) error {
+	if !t.enabled {
+		return nil
+	}
+
+	item := UniqueEvent{
+		Event: UniqueEventMetadata{
+			Component: component,
+			EventType: eventType,
+		},
+		Key: key,
+	}
+	t.uniqueEventBatcher.AddNoWait(item)
 	return nil
 }
+
+func (t *TelemetrySender) HandleCounters(batch []UniqueEvent) error {
+	if len(batch) == 0 {
+		return nil
+	}
+
+	logrus.Infof("Telemetry batch: %v", batch)
+
+	for _, item := range batch {
+		t.uniqueEventsCounter.IncrementCounter(item.Event.Component, item.Event.EventType, item.Key)
+	}
+
+	counts := t.uniqueEventsCounter.Get()
+	logrus.Infof("Sending telemetry snapshot: %v", counts)
+	timeUntilReset := t.lastSnapshotResetTime.Add(t.snapshotResetInterval)
+
+	if time.Now().After(timeUntilReset) {
```

Comment:
I think it should be at the end of the function 🤔

Reply:
100%, also I abort the reset if the send fail